provider.go

```diff
@@ -0,0 +1,95 @@
+package joohoi_acme_dns
```

Comment:
Might be best to stick with Go convention, that is the name of the folder, and no underscores:

```suggestion
package acmedns
```